src/nvme/types.h

```diff
@@ -5997,6 +5999,75 @@ enum nvme_apst_entry {
 	NVME_APST_ENTRY_ITPT_MASK = 0xffffff,
 };
 
+/**
+ * struct nvme_std_perf_attr - Standard performance attribute structure
+ * @rsvd0:	Reserved
+ * @r4karl:	Random 4 KiB average read latency
+ * @rsvd5:	Reserved
+ */
+struct nvme_std_perf_attr {
+	__u8 rsvd0[4];
+	__u8 r4karl;
+	__u8 rsvd5[4091];
+};
+
+/**
+ * struct nvme_perf_attr_id - Performance attribute identifier structure
+ * @id:	Performance attribute identifier
+ */
+struct nvme_perf_attr_id {
+	__u8 id[NVME_UUID_LEN];
+};
+
+/**
+ * struct nvme_perf_attr_id_list - Performance attribute identifier list structure
+ * @attrtyp:	Bits 7-3: Reserved
+ *		Bits 2-0: Attribute type
+ * @msvspa:	Maximum saveable vendor specific performance attributes
+ * @usvspa:	Unused saveable vendor specific performance attributes
+ * @rsvd3:	Reserved
+ * @id_list:	Performance attribute identifier list
+ * @rsvd1024:	Reserved
+ */
+struct nvme_perf_attr_id_list {
+	__u8 attrtyp;
+	__u8 msvspa;
+	__u8 usvspa;
+	__u8 rsvd3[13];
+	struct nvme_perf_attr_id id_list[1008];
+	__u8 rsvd1024[3072];
+};
+
+/**
+ * struct nvme_vs_perf_attr - Vendor specific performance attribute structure
+ * @paid:	Performance attribute identifier
+ * @rsvd16:	Reserved
+ * @attrl:	Attribute Length
+ * @vs:		Vendor specific
+ */
+struct nvme_vs_perf_attr {
+	struct nvme_perf_attr_id paid;
```

Comment:
Are you sure we want to model it this way? From the spec I would just use ` __u8 paid[NVME_UUID_LEN]` directly. 

Should `nvme_perf_attr_id` exist as it is not really directly as data struct defined in the spec? I'd like to stick as close as possible to the spec and don't come up with our own data structs. While I agree in this case here it would make sense (and it is cleaner), I think it would be better to stick to the spec. 

I can be convinced to do add it though. Thoughts? @keithbusch